src_python/habitat_sim/viewer/viewer.py

```diff
@@ -0,0 +1,1170 @@
+    # window height * TEXT_DELTA_FROM_CENTER in the y axis, as the text
+    # position defaults to the middle of the app window)
+    TEXT_DELTA_FROM_CENTER = 0.49
+
+    # font size of the magnum in-window display text that displays
+    # CPU and GPU usage info
+    DISPLAY_FONT_SIZE = 16.0
+
+    def __init__(self, sim_settings: Dict[str, Any]) -> None:
+        self.sim_settings: Dict[str, Any] = sim_settings
+
+        self.enable_batch_renderer: bool = self.sim_settings["enable_batch_renderer"]
+        self.num_env: int = (
+            self.sim_settings["num_environments"] if self.enable_batch_renderer else 1
+        )
+
+        # Compute environment camera resolution based on the number of environments to render in the window.
+        surface_size: Tuple[int, int] = (
+            self.sim_settings["window_width"],
+            self.sim_settings["window_height"],
+        )
+        grid_size: Tuple[int, int] = ReplayRenderer.environment_grid_size(self.num_env)
+        camera_resolution: Tuple[int, int] = (
+            int(surface_size[0] / grid_size[0]),
+            int(surface_size[1] / grid_size[1]),
+        )
+        self.sim_settings["width"] = camera_resolution[0]
+        self.sim_settings["height"] = camera_resolution[1]
+
+        configuration = self.Configuration()
+        configuration.title = "Habitat Sim Interactive Viewer"
+        configuration.size = surface_size
+        Application.__init__(self, configuration)
+        self.fps: float = 60.0
+
+        # draw Bullet debug line visualizations (e.g. collision meshes)
+        self.debug_bullet_draw = False
+        # draw active contact point debug line visualizations
+        self.contact_debug_draw = False
+        # cache most recently loaded URDF file for quick-reload
+        self.cached_urdf = ""
+
+        # set up our movement map
+        key = Application.KeyEvent.Key
+        self.pressed = {
+            key.UP: False,
+            key.DOWN: False,
+            key.LEFT: False,
+            key.RIGHT: False,
+            key.A: False,
+            key.D: False,
+            key.S: False,
+            key.W: False,
+            key.X: False,
+            key.Z: False,
+        }
+
+        # set up our movement key bindings map
+        key = Application.KeyEvent.Key
+        self.key_to_action = {
+            key.UP: "look_up",
+            key.DOWN: "look_down",
+            key.LEFT: "turn_left",
+            key.RIGHT: "turn_right",
+            key.A: "move_left",
+            key.D: "move_right",
+            key.S: "move_backward",
+            key.W: "move_forward",
+            key.X: "move_down",
+            key.Z: "move_up",
+        }
+
+        # Load a TrueTypeFont plugin and open the font file
+        self.display_font = text.FontManager().load_and_instantiate("TrueTypeFont")
+        self.display_font.open_file(
+            os.path.join(os.path.dirname(__file__), self.sim_settings["font_path"]),
+            13,
+        )
```

Comment:
@mosra had to replace the relative filepath with a configurable one which can be updated in consumer code. Generally we can't expect a `data` folder to be available from conda build, so we should consider how to handle this. 

Maybe we should include the font file with the python module source code instead?

Reply:
> Maybe we should include the font file with the python module source code instead?

I think that this would make sense.

Reply:
Yep, that makes sense -- when John did that originally we weren't sure what the use case would be, and hardcoding the relative path to `data/` was the most straightforward way.